tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -44,50 +45,36 @@ public string OtherArgs
     {
         get; private set;
     }
-
-    [JsonIgnore]
+
+    [ObservableProperty]
+    private bool _isPinned;

```

Comment:
I'm struggling with this one a bit.... first we've got the pinned property, and then potientially an "order index in the bar"... is this a property of an ExternalTool, or how it's represented in the bar? Or is it ok to have them all in one data structure.

Not advocating for a particular change now, but just something to keep in mind in the future.

Reply:
IsPinned is rightfully a property of ExternalTool, same as arguments - both are PI-specific configurations for the tool that the user gets to choose. We use the IsPinned user setting to do X, and the Args user setting to do Y. Not sure where the "order index" comes from?

Reply:
We chatted a bit about long term vision. I'm ok with this for now.